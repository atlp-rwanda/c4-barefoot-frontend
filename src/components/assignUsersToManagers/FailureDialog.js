import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import{ useDispatch, useSelector }from 'react-redux';
import { assignUsersFromQueue, cancelAllQueue } from '../../redux/actions/managerSelectedActions';

export default function FailureDialog(props) {
  const retryAssign = useSelector(state => state.addAssignActionToQueue);
  const dispatch = useDispatch(); // for resetting the real [user to manager] state
  const handleTryAgain = () => {
    assignUsersFromQueue(dispatch, retryAssign.pendingTasks);
  }
  const handleCancel = () => {
    cancelAllQueue(dispatch);
  }
  return (
    <Dialog
      open={props.open}
      onClose={()=>{}}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title" style={{color: '#EB5757'}}>{"Error!"}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description" style={{color: '#EB5757'}}>
          Could not assign user(s) to manager(s)
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleTryAgain} color="primary">
          TRY AGAIN
        </Button>
        <Button onClick={handleCancel} style={{color: '#333333'}}>
          CANCEL
        </Button>
      </DialogActions>
    </Dialog>
  );
}
