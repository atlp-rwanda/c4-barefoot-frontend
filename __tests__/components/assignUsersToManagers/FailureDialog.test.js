import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow , mount} from "enzyme";
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import FailureDialog from '../../../src/components/assignUsersToManagers/FailureDialog';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
let store;
store = mockStore({
  addAssignActionToQueue: { pendingTasks: undefined }
});

describe('Failure Dialog while assigning users to managers', () => {
    const props = { open: true };
    it('should be rendered without errors', () => {
      const failureDialog = shallow(
        <Provider store={store}>
          <FailureDialog {...props} />
        </Provider>
      );
      expect(toJson(failureDialog)).toMatchSnapshot();
    });
});
