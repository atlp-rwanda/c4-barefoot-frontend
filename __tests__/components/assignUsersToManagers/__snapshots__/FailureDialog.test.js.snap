// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Failure Dialog while assigning users to managers should be rendered without errors 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <FailureDialog
    open={true}
  />
</ContextProvider>
`;
